test/e2e/minscale_readiness_test.go

```diff
@@ -150,3 +158,19 @@ func waitForDesiredScale(t *testing.T, clients *test.Clients, deploymentName str
 		return cond(*deployment.Spec.Replicas), nil
 	})
 }
+
+func waitForDesiredEndpoints(t *testing.T, clients *test.Clients, endpointName string, cond func(int32) bool) error {
```

Comment:
I think we can even go as far as using this as `waitForDesiredScale` instead of adding these calls in addition.